core/templates/pages/topic-editor-page/rearrange-skills-in-subtopics-modal.controller.ts

```diff
@@ -65,38 +67,61 @@ angular.module('oppia').controller(
        * @param {SkillSummary} skillSummary - The summary of the skill that
        *    is to be moved.
        */
-      $scope.onMoveSkillStart = function(oldSubtopicId, skillSummary) {
-        $scope.skillSummaryToMove = skillSummary;
-        $scope.oldSubtopicId = oldSubtopicId ? oldSubtopicId : null;
+      ctrl.onMoveSkillStart = function(oldSubtopicId, skillSummary) {
+        ctrl.skillSummaryToMove = skillSummary;
+        ctrl.oldSubtopicId = oldSubtopicId ? oldSubtopicId : null;
       };
 
       /**
        * @param {string|null} newSubtopicId - The subtopic to which the
        *    skill is to be moved, or null if the destination is the
        *    uncategorized section.
        */
-      $scope.onMoveSkillEnd = function(newSubtopicId) {
-        if (newSubtopicId === $scope.oldSubtopicId) {
+      ctrl.onMoveSkillEnd = function(newSubtopicId) {
+        if (newSubtopicId === ctrl.oldSubtopicId) {
           return;
         }
 
         if (newSubtopicId === null) {
           TopicUpdateService.removeSkillFromSubtopic(
-            $scope.topic, $scope.oldSubtopicId, $scope.skillSummaryToMove);
+            ctrl.topic, ctrl.oldSubtopicId, ctrl.skillSummaryToMove);
         } else {
           TopicUpdateService.moveSkillToSubtopic(
-            $scope.topic, $scope.oldSubtopicId, newSubtopicId,
-            $scope.skillSummaryToMove);
+            ctrl.topic, ctrl.oldSubtopicId, newSubtopicId,
+            ctrl.skillSummaryToMove);
         }
         _initEditor();
       };
 
-      $scope.init = function() {
+
+      ctrl.updateSubtopicTitle = function(subtopicId) {
+        if (!SubtopicValidationService.checkValidSubtopicName(
+          ctrl.editableName)) {
+          ctrl.errorMsg = 'A subtopic with this title already exists';
+          return;
+        }
+
+        TopicUpdateService.setSubtopicTitle(
+          ctrl.topic, subtopicId, ctrl.editableName);
+        ctrl.editableName = '';
```

Comment:
Seems redundant because we are doing this inside `editNameOfSubtopicWithId` too.

Reply:
Yes fixed.